Pixel size should be assigned when importing refinement package

twagner9 opened the pull request:
# Description

This was an odd bug that took a little while to identify, but required only a single-line fix. When importing a refinement package using cisTEM parameters, the refinement being created for the new package was initialized using the pixel size from the particle info, but the particle info never had its pixel size value assigned from the imported parameters, so it was initialized to 0.

There were 2 symptoms: when attempting to do 3D refinement, the process would crash due to invalid SSNR values; and when displaying the various FSC values with the popup dialog on the refinement results panel, there was a clear issue in SSNR values, which would reach infinity:

![bad_fsc](https://github.com/user-attachments/assets/cac8f079-acb6-4c7f-9032-846a95f79992)

After assigning the particle info a pixel size from the first line of the parameters file, the issue was resolved, refinement could be run successfully:

![good_fsc](https://github.com/user-attachments/assets/12e2b66c-bfa2-49fa-ad89-3779d78a6c4e)


Fixes #523

# I have rebased my feature branch to be current with the master branch using to minimize conflicts and headaches

- [x] yes
- [ ] no

# Which compilers were tested

- [x] g++
- [x] icpc
- [ ] clang
- [ ] other (please specify)

# These changes are isolated to the

- [x] gui
- [ ] core library
- [ ] gpu core library
- [ ] program it modifies

# How has the functionality been tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Tested manually from GUI
- [ ] Tested manually from CLI
- [ ] Passed console tests
- [ ] Passed samples functional testing
- [ ] other (please specify)

# Checklist:

- [x] I have not changed anything that did not need to be changed
- [x] I have performed a self-review of my own code
- [x] I have commented my code, (***w.r.t. why***), particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/bHimes/cisTEM_docs) {Ok to pass for now}
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

bHimes reviewed (dismissed):
This looks fine, but it does leave the door open for similar future bugs, which are super common when a change is made to copypasta like this. 

If you want to take a crack at some template programming, I've outlined a more robust fix/ refactor to help prevent future bugs in #523

twagner9 commented:
> If you want to take a crack at some template programming, I've outlined a more robust fix/ refactor to help prevent future bugs in #523

I spent the last few workdays working on the template programming, and was mostly able to follow the proposed path, with a few differences:

- `std::unique_ptr` couldn't be used for the refinement package because subsequent UI processes rely on the newly created refinement package and refinement remaining in memory beyond the function's scope; using the smart pointer, it went out of scope when the import function went out of scope, which would cause a crash. Reopening the project would cause no problems because the package was simply being loaded into memory from the database and not going of scope.
- Removed some comment clutter and a 7 year old comment that seemed unlikely to be used.
-  Refactored `OnBinClick` in `RefinementPackageAssetPanel` within wxFormBuilder, which was a forgotten/leftover change from adding GUI resampling.

A special note on this last bit was that I did have to use a newer version of wxFormBuilder as I've recently received a new machine that is using Ubuntu 24.04, and was having troubles trying to get earlier versions working, and newer version .fbp files are, of course, _not_ backwards compatible with older versions, so if anyone is planning to make GUI updates, they'll have to 4.2.0 or a newer version, at least on the .fbp I changed. Fortunately, there are .deb packages for Ubuntu 20.04, 22.04, and 24.04, so I think this shouldn't be a problem? Those are found [on their release page](https://github.com/wxFormBuilder/wxFormBuilder/releases).

bHimes commented:
> > If you want to take a crack at some template programming, I've outlined a more robust fix/ refactor to help prevent future bugs in #523
> 
> @bHimes
> 
> I spent the last few workdays working on the template programming, and was mostly able to follow the proposed path, with a few differences:
> 
>     * `std::unique_ptr` couldn't be used for the refinement package because subsequent UI processes rely on the newly created refinement package and refinement remaining in memory beyond the function's scope; using the smart pointer, it went out of scope when the import function went out of scope, which would cause a crash. Reopening the project would cause no problems because the package was simply being loaded into memory from the database and not going of scope. (This was what took me awhile to realize).
> 
>     * Removed some comment clutter and a 7 year old comment that seemed unlikely to be used.
> 
>     * Refactored `OnBinClick` in `RefinementPackageAssetPanel` within wxFormBuilder, which was a forgotten/leftover change from adding GUI resampling.
> 
> 
> A special note on this last bit was that I did have to use a newer version of wxFormBuilder as I've recently received a new machine that is using Ubuntu 24.04, and was having troubles trying to get earlier versions working, and newer version .fbp files are, of course, _not_ backwards compatible with older versions, so if anyone is planning to make GUI updates, they'll have to 4.2.0 or a newer version, at least on the .fbp I changed. Fortunately, there are .deb packages for Ubuntu 20.04, 22.04, and 24.04, so I think this shouldn't be a problem? Those are found [on their release page](https://github.com/wxFormBuilder/wxFormBuilder/releases).

@twagner9 This all looks good, nicely done. I'm fine with the code changes as far as the review goes. Regarding the change to the wxformbuilder version; I'm okay changing the toolchain, but it needs to work for everyone.

- As long as we can get 4.2 to run fine inside the devcontainer, then the PR is good to go. You should be able to test that in an existing copy of the container. I'll send an email with a few details.
- Alternatively, I'm not sure why you wouldn't have been able to use the current version of wxformbuilder from inside the container, even being on Ubuntu 24. Did you try this?

twagner9 commented:
>Alternatively, I'm not sure why you wouldn't have been able to use the current version of wxformbuilder from inside the container, even being on Ubuntu 24. Did you try this?

I was unaware that formbuilder was on the container; I will give this a try.
